### PR TITLE
frogmouth: update from xdg to xdg-base-dirs

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

Previously, running `frogmouth` resulted in a stack trace where `frogmouth` was unable to find `xdg` and could not start.

Now, the two places where `xdg` is imported are patched to use `xdg_base_dirs`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by philiptaron** [comment]
@figsoda Here's an update/fix for you, as the maintainer of this package. 💌 

I'll work to make the correct fix upstream.

**Comment by philiptaron** [comment]
Oh, I see you had already made this patch but it got rejected upstream. https://github.com/Textualize/frogmouth/pull/59. Sad! Well, at least we can have nice things on NixOS.

File: pkgs/tools/text/frogmouth/default.nix [hunk 1]
```diff
@@ -15,6 +15,8 @@ python3.pkgs.buildPythonApplication rec {
     hash = "sha256-TMCeIwMEoNDQV9iue4XrdYQMmwofXDvdMMLTZKILQ9Q=";
   };
 
+  patches = [ ./use-xdg-base-dirs.patch ];
```
Comment (eclairevoyant) [inline pkgs/tools/text/frogmouth/default.nix:18]: Please add a comment as to why this is needed. Also can thes patch be upstreamed, or can we fetch the patch from upstream?

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/frogmouth/default.nix (hunk 1 above)
Comment (philiptaron) [reply pkgs/tools/text/frogmouth/default.nix:18]: I can fetch the patch from upstream (@figsoda's patch https://github.com/Textualize/frogmouth/pull/59 has rotted; my PR https://github.com/Textualize/frogmouth/pull/85 is still able to be applied.)

It looks like upstream is not disposed to merge it, though, for reasons involving the metadata on `xdg-base-dirs`.

Comment (philiptaron) [reply pkgs/tools/text/frogmouth/default.nix]: I've pushed a new commit that just uses `sed` on the relevant files.

**Comment by philiptaron** [comment]
I've replaced the previous patch with a one-line `sed` command.

**Review by figsoda — APPROVED** [review]
thanks!

**Pull request merged**